nbclient/client.py

```diff
@@ -805,11 +865,13 @@ async def async_execute_cell(
             self.allow_errors or "raises-exception" in cell.metadata.get("tags", [])
         )
 
+        await run_hook(self.on_cell_start, cell=cell, cell_index=cell_index)
```

Comment:
while we are at it, perhaps there could be a hook at the very start of this function, before even skipping non-executing cells, like `on_cell_visit`

Reply:
Does it make sense to just move `on_cell_start` to the start instead of creating a new hook?

Reply:
That would make sense, and we know if the cell is a Markdown cell with `cell.cell_type`.
But then maybe add `on_cell_execute` right before the execution? It won't be called if the cell is not a code cell or if the cell is skipped because of a tag.

Reply:
I moved `on_cell_start` towards the top so it'll execute for non-code cells and added  a new hook called `on_cell_execute`